canonical-web-and-design/react-components
# Handle adjusting the tooltip so that it remains on screen

## Done
- Added useFitsScreen, useListener, usePrevious and useThrottle hooks.
- Handle adjusting the tooltip so that it remains on screen.

## QA

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Navigate to the Tooltip examples.
- Check that the follow mouse example follows the mouse.
- Resize your window to different sizes. The tooltip should adjust so that it remains on the screen (except where it can't).
